### ISSUE TITLE
FauxStatement, passing null or custom Object.

### ISSUE THREAD

**Issue by hitech95** [DataPubAPI/DataPubAPI#5]
I gave a quick look at the code, one of the things that caught my eye is that to create the statement usae setObject (FauxStatement). I have read the documentation of Java and I read this thing:

> Note: Not all databases allow for a non-typed Null to be sent to the backend. For maximum portability, > the setNull or the setObject(int parameterIndex, Object x, int sqlType) method should be used instead > of setObject(int parameterIndex, Object x).

> Note: This method throws an exception if there is an ambiguity, for example, if the object is of a class > implementing more than one of the interfaces named above.

Bye, hitech95

**Comment by FerusGrim** [comment]
My API cannot handle custom objects any better than any SQL database.

If you're using a custom object, it must be passed in a way that the database can handle. As far as null goes, I'll look into it.

**Comment by hitech95** [comment]
The problem is that I can pass any object to your method.

**Comment by FerusGrim** [comment]
And it would fail. So you shouldn't do that.

Unless you have a proposed solution?

**Comment by Xemiru** [comment]
Not necessarily that you can pass any object; the method will fail on its own and throw an exception if an attempt to compile the statement was made. However, the improvement that I'm thinking he's suggesting is that we implement a safecheck for attempts to pass objects as replacements, checking the type of the statements and if they're applicable for the method. Under the perspective of idiot-proofing, this will help prevent any unintended side-effects that would result from attempting to call the method with bad parameters, and will help in debugging the problem thrown should this situation occur.

tl;dr he wants type safety (`instanceof` checks) on the method instead of just freely allowing objects to be thrown at it

**Comment by hitech95** [comment]
@Xemiru Yup

**Comment by Xemiru** [comment]
Also, custom objects should be implementing `SQLData` if anything, otherwise it shouldn't be accepted as a custom type.

**Comment by Xemiru** [comment]
Fixed in f3821e0.

**Comment by Xemiru** [comment]
Fixed in 44c3343.
